src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/RSSAnalyzer.scala

```diff
@@ -13,21 +13,40 @@ class RSSAnalyzer extends Analyzer[RSSEntry] with Serializable with AnalysisDefa
 
   override def toSchema(item: RSSEntry, locationFetcher: LocationFetcher, imageAnalyzer: ImageAnalyzer): ExtendedDetails[RSSEntry] = {
     val body = readDescription(item)
+    val url = getSourceUrlFromItem(item)
     ExtendedDetails(
-      eventid = s"RSS.${item.uri}",
+      eventid = s"RSS.${url.getOrElse(item.uri)}",
       sourceeventid = item.uri,
       eventtime = item.publishedDate,
       body = body,
       imageurl = None,
       title = item.title,
       externalsourceid = item.source.uri,
       pipelinekey = "RSS",
-      sourceurl = item.uri,
+      sourceurl = url.getOrElse(""),
       sharedLocations = List(),
       original = item
     )
   }
 
+  private[analyzer] def getSourceUrlFromItem(item: RSSEntry): Option[String] = {
+    getSourceUrl(item.uri) match {
+      case Some(url) => Some(url)
+      case _ => {
+        if (item.links == null || item.links.isEmpty) None
+        else getSourceUrl(item.links.head.href)
+      }
+    }
+  }
+
+  private[analyzer] def getSourceUrl(url: String): Option[String] = {
+    try {
+      Option(new URL(url).toURI.toASCIIString)
+    } catch {
+      case _: Exception => None
```

Comment:
What will this throw? Should at least catch nonfatal instead. If you use Try().toOption, catching nonfatal only will be handled automatically :)

Reply:
Oh, I didn't know about that - will do (separate PR?) To answer your question, this will most often throw something like a java.net.MalformedURLException.